From 72259ae325d67e93d3b89c5c58e9290041e02add Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 15 Sep 2025 23:17:42 +0800
Subject: [PATCH 11/14] DrawTargetCG.cpp: macros should be SDK-conditional, not
 arch-conditional

Experimental; if does not work on 10.6, adjust the value.
---
 gfx/2d/DrawTargetCG.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/gfx/2d/DrawTargetCG.cpp b/gfx/2d/DrawTargetCG.cpp
index 754fa35c2a..1bed1efaab 100644
--- a/gfx/2d/DrawTargetCG.cpp
+++ b/gfx/2d/DrawTargetCG.cpp
@@ -172,7 +172,7 @@ DrawTargetCG::~DrawTargetCG()
 DrawTargetType
 DrawTargetCG::GetType() const
 {
-#if !defined(__ppc__)
+#if defined(__APPLE__) && (MAC_OS_X_VERSION_MIN_REQUIRED >= 1060)
   return GetBackendType() == BackendType::COREGRAPHICS_ACCELERATED ?
            DrawTargetType::HARDWARE_RASTER : DrawTargetType::SOFTWARE_RASTER;
 #else
@@ -183,7 +183,7 @@ DrawTargetCG::GetType() const
 BackendType
 DrawTargetCG::GetBackendType() const
 {
-#if defined(MOZ_WIDGET_COCOA) && !defined(__ppc__)
+#if defined(MOZ_WIDGET_COCOA) && (MAC_OS_X_VERSION_MIN_REQUIRED >= 1060)
   // It may be worth spliting Bitmap and IOSurface DrawTarget
   // into seperate classes.
   if (GetContextType(mCg) == CG_CONTEXT_TYPE_IOSURFACE) {
@@ -200,7 +200,7 @@ already_AddRefed<SourceSurface>
 DrawTargetCG::Snapshot()
 {
   if (!mSnapshot) {
-#if defined(MOZ_WIDGET_COCOA) && !defined(__ppc__)
+#if defined(MOZ_WIDGET_COCOA) && (MAC_OS_X_VERSION_MIN_REQUIRED >= 1060)
     if (GetContextType(mCg) == CG_CONTEXT_TYPE_IOSURFACE) {
       return MakeAndAddRef<SourceSurfaceCGIOSurfaceContext>(this);
     }
@@ -1692,7 +1692,7 @@ DrawTargetCG::Init(BackendType aType,
 
   mSize = aSize;
 
-#ifdef MOZ_WIDGET_COCOA
+#if defined(MOZ_WIDGET_COCOA) && (MAC_OS_X_VERSION_MIN_REQUIRED >= 1060)
   if (aType == BackendType::COREGRAPHICS_ACCELERATED) {
     RefPtr<MacIOSurface> ioSurface = MacIOSurface::CreateIOSurface(aSize.width, aSize.height);
     mCg = ioSurface->CreateIOSurfaceContext();
@@ -1860,7 +1860,7 @@ DrawTargetCG::CreatePathBuilder(FillRule aFillRule) const
 void*
 DrawTargetCG::GetNativeSurface(NativeSurfaceType aType)
 {
-#if defined(MOZ_WIDGET_COCOA) && !defined(__ppc__)
+#if defined(MOZ_WIDGET_COCOA) && (MAC_OS_X_VERSION_MIN_REQUIRED >= 1060)
   if ((aType == NativeSurfaceType::CGCONTEXT && GetContextType(mCg) == CG_CONTEXT_TYPE_BITMAP) ||
       (aType == NativeSurfaceType::CGCONTEXT_ACCELERATED && GetContextType(mCg) == CG_CONTEXT_TYPE_IOSURFACE)) {
     return mCg;
-- 
2.50.1

